roc_jni/src/main/cpp/receiver.cpp

```diff
@@ -73,37 +78,58 @@ JNIEXPORT jlong JNICALL Java_org_rocstreaming_roctoolkit_Receiver_open(JNIEnv *
     return (jlong) receiver;
 }
 
+JNIEXPORT void JNICALL Java_org_rocstreaming_roctoolkit_Receiver_setMulticastGroup(JNIEnv * env, jobject thisObj,
+                    jlong receiverPtr, jint slot, jint interface, jstring jip) {
+    roc_receiver* receiver    = NULL;
+    const char*   ip;
+
+    receiver = (roc_receiver*) receiverPtr;
+    ip = env->GetStringUTFChars(jip, 0);
+
+    if (roc_receiver_set_multicast_group(receiver, (roc_slot) slot, (roc_interface) interface, ip) != 0) {
+        env->ReleaseStringUTFChars(jip, ip);
+        jclass exceptionClass = env->FindClass(EXCEPTION);
+        env->ThrowNew(exceptionClass, "Couldn't set multicast group");
+        return;
+    }
+    env->ReleaseStringUTFChars(jip, ip);
+}
+
 JNIEXPORT void JNICALL Java_org_rocstreaming_roctoolkit_Receiver_bind(JNIEnv * env, jobject thisObj, jlong receiverPtr,
-                    jint type, jint protocol, jobject jaddress) {
+                    jint slot, jint interface, jobject jendpoint) {
     roc_receiver*   receiver    = NULL;
-    roc_address     address;
+    roc_endpoint*   endpoint;
     int             port        = 0;
 
     receiver = (roc_receiver*) receiverPtr;
 
-    if (address_unmarshall(env, &address, jaddress) != 0) {
+    if (endpoint_unmarshal(env, &endpoint, jendpoint) != 0) {
+        roc_endpoint_deallocate(endpoint);
```

Comment:
At this point, we don't know if endpoint was (de)allocated or not. So we should not call roc_endpoint_deallocate() here.